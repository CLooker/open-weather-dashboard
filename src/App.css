body {
  min-width: 1000px;
  margin: 0 50px;
}

ul {
  list-style-type: none;
}

img {
  width: 50px;
  height: 50px;
}

.component-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.alert-and-quality-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 20px;
}

.unit-switcher {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  margin-bottom: 20px;
}

.current-weather-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.current-weather {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  border: 1px solid lightgrey;
  border-radius: 10px;
}

.weather * {
  padding: 20px;
}

.forecast {
  display: flex;
  justify-content: center;
}

.forecast * {
  min-width: 20px;
}

.forecast-list > p:first-child {
  min-width: 200px;
}

.forecast-list {
  display: flex;
  flex-direction: column;
}

.weather-and-icon {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.five-day-forecast {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 1px solid lightgrey;
  margin: 15px 50px 15px 50px;
  border-radius: 10px;
}

.forecast {
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
}

.forecast-list div {
  display: flex;
  align-items: center;
}

.air-quality-report-container {
  border: solid 1px lightgrey;
  border-radius: 10px;
  margin: 10px;
}

.air-quality-report > div:first-child {
  display: flex;
  justify-content: center;
  align-items: center;
}

.chem-container {
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
}

.pollutant-stats {
  margin: 10px;
}

.weather-alert-container {
  border: solid 1px lightgrey;
  border-radius: 10px;
  margin: 10px;
}

.weather-alert-container-title-icon {
  display: flex;
  justify-content: center;
}

.weather-alert-container-title-icon h3 {
  text-align: center;
}

.alert-container {
  display: flex;
  align-items: baseline;
  justify-content: flex-start;
}

.create-weather-alert-container {
  margin-right: 15px;
}

.create-weather-alert {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.create-weather-alert-title-icon {
  display: flex;
  justify-content: center;
  align-items: center;
}

.create-weather-alert-title-icon img {
  width: 25px;
  height: 25px;
}

.create-weather-alert form {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.create-weather-alert form div {
  display: flex;
}

.create-weather-alert form input:last-child {
  margin-top: 10px;
}

.get-weather-alerts {
  margin-right: 5px;
}

.unit-switcher input {
  width: 50px;
}

div.forecast-list,
div.forecast-list p {
  display: flex;
  justify-content: center;
  align-items: center;
}
