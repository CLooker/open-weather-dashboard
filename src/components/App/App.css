body {
  min-width: 1000px;
  overflow: auto;
  overflow-y: scroll;
}

ul {
  list-style-type: none;
}

img {
  width: 50px;
  height: 50px;
}

.app-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.alert-and-quality-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin: -10px 0 10px 0;
}

.unit-switcher {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  margin: 20px;
}

.current-weather-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.current-weather {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  border: 1px solid lightgrey;
  border-radius: 10px;
}

.weather * {
  padding: 0 20px;
}

.forecasts {
  display: flex;
  justify-content: center;
}

.forecast-list:first-child {
  min-width: 200px;
}

.forecast-list {
  text-align: center;
  display: flex;
  flex-direction: column;
}

.five-day-forecast {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid lightgrey;
  margin: 15px 25px;
  border-radius: 10px;
}

.forecast {
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
}

.forecast-list div {
  display: flex;
  align-items: center;
  justify-content: center;
}

.air-quality-report-icon-and-text-container {
  border: solid 1px lightgrey;
  border-radius: 10px;
  margin: 10px;
  padding: 15px 0 0;
}

.air-quality-report-icon-and-text > div:first-child {
  display: flex;
  justify-content: center;
  align-items: center;
}

.pollutant-reports-container {
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
}

.pollutant-report {
  margin: 10px;
}

.pollutant-report * {
  display: flex;
}

.weather-alerts-container {
  border: solid 1px lightgrey;
  border-radius: 10px;
  margin: 10px;
}

.weather-alerts-container > * {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 15px 0 0;
}

.alert-container {
  display: flex;
  align-items: baseline;
  justify-content: flex-start;
}

.create-weather-alerts-container {
  margin-right: 15px;
}

.create-weather-alert {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.create-weather-alert-title-icon {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px 0;
}

.create-weather-alert-title-icon img {
  width: 25px;
  height: 25px;
}

.create-weather-alert form {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.create-weather-alert form div {
  display: flex;
}

.create-weather-alert form input:last-child {
  margin-top: 10px;
}

.get-weather-alerts {
  margin: 0 5px;
  padding: 0 0 0 10px;
  position: relative;
  bottom: 10px;
}

.unit-switcher input {
  width: 50px;
}

div.forecast-list,
div.forecast-list p {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 15px;
}

h4 {
  font-size: 15px;
}

div.forecast-list img {
  padding: 0 5px 0 0;
}

div.weather div {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.five-day-forecast-title {
  margin: 20px 0 -1px;
  padding: 0 0 10px;
}

.pollutant-report-info {
  display: flex;
}

.pollutant-report h5 {
  margin-right: 5px;
}

.pollutant-report p {
  font-size: 15px;
}

.pollutant-report div {
  display: flex;
  flex-direction: column;
  align-items: left;
}

.pollutant-report * {
  margin: 0;
  padding: 0;
}

.create-weather-alert-title-icon p {
  font-size: 15px;
}

.get-weather-alerts p {
  font-weight: bold;
  font-size: 12px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}
