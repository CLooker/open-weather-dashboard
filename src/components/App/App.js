import React, { Component } from 'react';
import styled from 'styled-components';
import AirQualityReport from '../AirQualityReport';
import CurrentWeather from '../CurrentWeather';
import FiveDayForecast from '../FiveDayForecast';
import UnitsSwitcher from '../UnitsSwitcher';
import WeatherAlerts from '../WeatherAlerts';

const AppWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const Wrapper = styled.div`
  display: flex;
  justify-content: space-around;
  min-width: 927px;
`;

export default class App extends Component {
  render() {
    return (
      <AppWrapper>
        <h1>Current Weather and Forecast in Rome</h1>
        <UnitsSwitcher />
        <CurrentWeather />
        <FiveDayForecast />
        <Wrapper>
          <WeatherAlerts />
          <AirQualityReport />
        </Wrapper>
      </AppWrapper>
    );
  }
}
