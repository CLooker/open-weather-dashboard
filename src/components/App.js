import React, { Component } from 'react';
import WeatherAndForecastLocation from './WeatherAndForecastLocation';
import UnitsSwitcher from './UnitsSwitcher';
import CurrentWeather from './CurrentWeather';
import FiveDayForecast from './FiveDayForecast';
import AirQualityReport from './AirQualityReport';
import WeatherAlertContainer from './WeatherAlertContainer';
import '.././App.css';

export default class App extends Component {
  state = {
    name: null,
    units: 'F'
  };

  setName = name => this.setState({ name });

  setUnits = units => this.setState({ units });

  render() {
    return (
      <div className="component-container">
        <WeatherAndForecastLocation name={this.state.name} />
        <UnitsSwitcher
          name={this.state.name}
          setUnits={this.setUnits}
          units={this.state.units}
        />
        <CurrentWeather setName={this.setName} units={this.state.units} />
        <FiveDayForecast name={this.state.name} units={this.state.units} />
        <div className="alert-and-quality-container">
          <WeatherAlertContainer />
          <AirQualityReport />
        </div>
      </div>
    );
  }
}
