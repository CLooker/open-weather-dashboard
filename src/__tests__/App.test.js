import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../components/App';

describe('App', () => {
  let component;

  beforeEach(() => (component = shallow(<App />)));

  describe('initialization', () => {
    it('should render correctly', () => {
      expect(component.debug()).toMatchSnapshot();
    });

    it('should have a particular state', () => {
      const actual = component.state();
      const expected = {
        name: null,
        units: 'F'
      };
      expect(actual).toEqual(expected);
    });

    describe('passing of state props to children', () => {
      let mountedComponent;

      beforeEach(() => {
        mountedComponent = mount(<App />);
      });

      describe('props passed to `UnitsSwitcher`', () => {
        let actual;

        beforeEach(() => {
          actual = mountedComponent.find('UnitsSwitcher').props();
        });

        it('should pass the correct props', () => {
          const expected = {
            name: null,
            units: 'F'
          };
          expect(actual).toEqual(expect.objectContaining(expected));
        });

        it('should pass a `setUnits` method', () => {
          expect(actual.setUnits).toEqual(expect.any(Function));
        });
      });

      describe('props passed to `CurrentWeather`', () => {
        let actual;

        beforeEach(() => {
          actual = mountedComponent.find('CurrentWeather').props();
        });

        it('should pass the correct props', () => {
          const expected = {
            units: 'F'
          };
          expect(actual).toEqual(expect.objectContaining(expected));
        });

        it('should pass a `setName` method', () => {
          expect(actual.setName).toEqual(expect.any(Function));
        });
      });

      describe('props passed to `FiveDayForecast`', () => {
        let actual;

        beforeEach(() => {
          actual = mountedComponent.find('FiveDayForecast').props();
        });

        it('should pass the correct props', () => {
          const expected = {
            name: null,
            units: 'F'
          };
          expect(actual).toEqual(expected);
        });
      });
    });
  });

  describe('methods', () => {
    describe('`setName`', () => {
      it('should set the `name` prop in state', () => {
        const name = 'Chicago';
        component.instance().setName(name);
        const actual = component.state().name;
        const expected = name;
        expect(actual).toBe(expected);
      });
    });

    describe('`setUnits`', () => {
      it('should set the `units` prop in state', () => {
        const units = 'C';
        component.instance().setUnits(units);
        const actual = component.state().units;
        const expected = units;
        expect(actual).toBe(expected);
      });
    });
  });
});
