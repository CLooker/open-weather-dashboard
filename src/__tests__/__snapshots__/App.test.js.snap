// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App initialization should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WeatherAndForecastLocation
          name={null}
        />,
        <UnitsSwitcher
          name={null}
          setUnits={[Function]}
          units="F"
        />,
        <CurrentWeather
          setName={[Function]}
          units="F"
        />,
        <FiveDayForecast
          name={null}
          units="F"
        />,
        <div
          className="alert-and-quality-container"
        >
          <WeatherAlertContainer />
          <AirQualityReport />
        </div>,
      ],
      "className": "component-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": null,
          "setUnits": [Function],
          "units": "F",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "setName": [Function],
          "units": "F",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": null,
          "units": "F",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WeatherAlertContainer />,
            <AirQualityReport />,
          ],
          "className": "alert-and-quality-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WeatherAndForecastLocation
            name={null}
          />,
          <UnitsSwitcher
            name={null}
            setUnits={[Function]}
            units="F"
          />,
          <CurrentWeather
            setName={[Function]}
            units="F"
          />,
          <FiveDayForecast
            name={null}
            units="F"
          />,
          <div
            className="alert-and-quality-container"
          >
            <WeatherAlertContainer />
            <AirQualityReport />
          </div>,
        ],
        "className": "component-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": null,
            "setUnits": [Function],
            "units": "F",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setName": [Function],
            "units": "F",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": null,
            "units": "F",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WeatherAlertContainer />,
              <AirQualityReport />,
            ],
            "className": "alert-and-quality-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
